business/dashboards.go

```diff
@@ -39,12 +39,20 @@ func NewDashboardsService() *DashboardsService {
 	if nsLabel == "" {
 		nsLabel = "kubernetes_namespace"
 	}
+
+	// Overwrite Custom dashboards defined at Namespace level
+	builtInDashboards := cfg.CustomDashboards.OrganizeByName()
+	nsDashboards := dashboards.GetNamespaceMonitoringDashboards(namespace.Annotations)
```

Comment:
See my earlier comment about wanting to log the namespace if this function `GetNamespaceMonitoringDashboards` results in an error. This appears to be the only place where this function is called. So we should either (a) pass in `namespace` as a new parameter to this function or (b) bubble up the err from this function and log that error along with `namespace` so the user can easily find the bad annotation in order to fix it.

Reply:
+1
I'm keeping the original design, so, if some problem happens at namespace level, Kiali logs the error with the namespace (thanks for the catch) and it will fall with the default one.

My first draft had that but I refactored to not create a circular dependency here, but yes, passing a namespace parameter is necessary.